### PR TITLE
Merge dev/svcutil to master

### PR DESCRIPTION
**Pull request by shmao**
The PR is to merge the dev/svcutil branch into the master branch.

### CODE REVIEW

**Comment by shmao** [comment]
I'm going to sent a PR from dev/svcutil directly

**Comment by shmao** [comment]
To revieweres: only the last two commits are new changes. The other commits had been reviewed and merged in dev/svcutil branch.

File: src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/CommandLineParser.cs [hunk 1]
```diff
@@ -0,0 +1,248 @@
+//-----------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//-----------------------------------------------------------------------------
```
Comment (zhenlan) [inline src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/CommandLineParser.cs]: Can we please update the header of all .cs files with the MIT license header? Did we run code formatting already? It's okay to do that in a separate PR. I just call it out.

File: src/svcutilcore/dir.props [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="..\dir.props" />
+  <PropertyGroup>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <AssemblyKey>Open</AssemblyKey>
+    <IsNETCoreApp>true</IsNETCoreApp>
+    <IsNETCoreAppRef>true</IsNETCoreAppRef>
```
Comment (zhenlan) [inline src/svcutilcore/dir.props]: I am wondering if `IsNETCoreAppRef` should be false as svcutil is a tool and does not expose any APIs, but I may not understand `IsNETCoreAppRef` correctly.

File: src/svcutilcore/src/svcutilcore.csproj [hunk 3]
```diff
@@ -0,0 +1,150 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <AssemblyName>svcutilcore</AssemblyName>
```
Comment (zhenlan) [inline src/svcutilcore/src/svcutilcore.csproj]: Does this need to be something like `dotnet-svcutil`?

File: src/svcutilcore/src/svcutilcore.csproj [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    <Reference Include="System.Windows" />
+    <Reference Include="System.Xml" />
+    <Reference Include="System.Xml.Linq" />
+    <Reference Include="System.Xml.ReaderWriter" />
+    <Reference Include="System.Xml.Serialization" />
+    <Reference Include="System.Xml.XDocument" />
+    <Reference Include="System.Xml.XPath" />
+    <Reference Include="System.Xml.XPath.XDocument" />
+    <Reference Include="System.Xml.XmlDocument" />
+    <Reference Include="System.Xml.XmlSerializer" />
```
[the remark this answers is not in the document]
Comment (shmao) [reply src/svcutilcore/src/svcutilcore.csproj]: I've removed unnecessary references. Thanks.

File: src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/CommandLineParser.cs (hunk 1 above)
Comment (shmao) [reply src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/CommandLineParser.cs]: I'll run CodeFormatter in a separate PR.
